tests/transformers/test_transformers.py

```diff
@@ -658,6 +658,37 @@ def test_leaves_data_unchanged(self):
     def test_raises_error_on_nonexistent_source_name(self):
         assert_raises(KeyError, Rename, self.stream, {'Z': 'features'})
 
+    def test_raises_on_invalid_kwargs(self):
+        assert_raises(ValueError, Rename, self.stream,
+                      {'X': 'features'}, on_non_existent='foo')
+
+    def test_name_clash(self):
+        assert_raises(KeyError, Rename, self.stream, {'X': 'y'})
+
+    def test_name_swap(self):
+        assert_equal(Rename(self.stream,
+                            {'X': 'y', 'y': 'X'},
+                            on_non_existent='ignore').sources,
+                     ('y', 'X'))
+
+    def test_raises_on_not_one_to_one(self):
+        assert_raises(ValueError, Rename, self.stream, {'X': 'features'},
+                      on_non_existent='foo')
```

Comment:
Yep, my bad. Will fix that shortly.

On Fri, May 6, 2016, 4:10 PM dmitriy-serdyuk <notifications@github.com>
wrote:

> In tests/transformers/test_transformers.py
> <https://github.com/mila-udem/fuel/pull/355#discussion_r62384186>:
>
> > +    def test_raises_on_invalid_kwargs(self):
> > +        assert_raises(ValueError, Rename, self.stream,
> > +                      {'X': 'features'}, on_non_existent='foo')
> > +
> > +    def test_name_clash(self):
> > +        assert_raises(KeyError, Rename, self.stream, {'X': 'y'})
> > +
> > +    def test_name_swap(self):
> > +        assert_equal(Rename(self.stream,
> > +                            {'X': 'y', 'y': 'X'},
> > +                            on_non_existent='ignore').sources,
> > +                     ('y', 'X'))
> > +
> > +    def test_raises_on_not_one_to_one(self):
> > +        assert_raises(ValueError, Rename, self.stream, {'X': 'features'},
> > +                      on_non_existent='foo')
>
> Still, this test tests some other stuff. Copy-paste?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mila-udem/fuel/pull/355/files/03d99b1f4e4a1791fa99d0e945ce0c30e8d72e22#r62384186>
>